third_party/txt/src/txt/paragraph.h

```diff
@@ -55,6 +55,31 @@ class Paragraph {
 
   enum Affinity { UPSTREAM, DOWNSTREAM };
 
+  // Options for various types of bounding boxes provided by
+  // GetRectsForRange(...).
+  // These options can be individually enabled, for example:
+  //
+  //   (RectStyle::kTight | RectStyle::kExtendEndOfLine)
+  //
+  // provides tight bounding boxes and extends the last box per line to the end
+  // of the layout area.
+  enum RectStyle {
+    kNone = 0x0,  // kNone cannot be combined with |.
+
+    // Provide tight bounding boxes that fit heights per span. Otherwise, the
+    // heights of spans are the max of the heights of the line the span belongs
+    // in.
+    kTight = 0x1,
+
+    // TODO(garyq): Implement line spacing inclusion/exclusion
+    // Include the height of the extra line spacing in the rects.
+    kIncludeLineSpacing = 0x2,
+
+    // TODO(garyq): Implement extension to end of line
+    // Extend the last rect of each line to the end of the line (macOS style)
+    kExtendEndOfLine = 0x4
+  };
```

Comment:
we should probably not actually include the values we don't support yet. We can have a comment saying that this is why we are using an enum, though.

Reply:
I have a followup PR that should be up soon that does implement it on the libtxt side.

Reply:
i would add these in the follow-up pr then :-)

Reply:
Sounds good!